src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3249,6 +3253,24 @@ class SymbolTableVisitor : public CommonVisitor<SymbolTableVisitor> {
                 }
             }
         }
+
+        std::string sym_name = x.m_name;
+        if (overload) {
+            std::string overload_number;
+            if (overload_defs.find(sym_name) == overload_defs.end()){
+                overload_number = "0";
+                Vec<ASR::symbol_t *> v;
+                v.reserve(al, 1);
+                overload_defs[sym_name] = v;
+            } else {
+                overload_number = std::to_string(overload_defs[sym_name].size());
+            }
+            sym_name = "__lpython_overloaded_" + overload_number + "__" + sym_name;
+        }
+        if (parent_scope->get_scope().find(sym_name) != parent_scope->get_scope().end()) {
+            throw SemanticError("Procedure, " + std::string(x.m_name) +  " is already defined", x.base.base.loc);
```

Comment:
I would remove the comma, and also in Python they are called "functions" I think:
```suggestion
            throw SemanticError("Function " + std::string(x.m_name) +  " is already defined", x.base.base.loc);
```

Reply:
Done.